Decouples Websocket code from HTML

nkinkade opened the pull request:
Decouples Websocket code from HTML so external sites can use Websocket code with their own HTML frontend (to some extent). The basic idea behind these changes, notwithstanding some style changes like consistently using single quotes, is that the core Websocket scripts can be used by other front-end implementations other than fakewww. The principal motivation for doing this is so that the M-Lab website can use the official NDT Websocket code located in the NDT project repository, while still maintaining control over the user interface so that it fits better stylistically, and also so that it can be embedded using HTTPS.

I have compiled this and tested that it works as expected. There should be no changes at all to how the Websocket test currently functions or looks when this is running via fakewww.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Review on Reviewable"/>](https://reviewable.io/reviews/m-lab/ndt/46)
<!-- Reviewable:end -->

mtlynch commented on HTML5-frontend/ndt-wrapper.js:
```diff
@@ -57,19 +51,19 @@ NDTWrapper.prototype.reset = function () {
 };
 
-NDTWrapper.prototype.run_test = function () {
+NDTWrapper.prototype.run_test = function (basePath) {
```
Can we document what `basePath` is?

mtlynch commented on HTML5-frontend/ndt-wrapper.js:
```diff
@@ -122,8 +117,14 @@ NDTWrapper.prototype.run_test = function () {
+      } else {
+        console.log('Could not determine a suitable NDT server. Aborting.');
+        return false;
+      }
```
We probably shouldn't even be in this function if we failed to find a suitable NDT server. If we're here and `this._hostname` is null, it means the calling code is wrong, so suggest re-wording to something like "No server hostname specified. Aborting."

Also, this is a style preference, so it's up to you, but I generally find code more readable when it's structured like:

```
if (!this._hostname) {
  console.log('whatever');
  return false;
}
this.client = new NDTjs(this._hostname, this._port, this._path,
                        this.callbacks, this._update_interval);
this.client.startTest();
```

mtlynch commented on HTML5-frontend/ndt-wrapper.js:
```diff
@@ -122,8 +117,14 @@ NDTWrapper.prototype.run_test = function () {
+      if (this._hostname) {
+        this.client = new NDTjs(this._hostname, this._port, this._path,
+                                this.callbacks, this._update_interval);
+        this.client.startTest();
```
Do we have to save `this.client`? Could we simplify to:

```
(new NDTjs(this._hostname, this._port, this._path, this.callbacks, this._update_interval)).startTest();

mtlynch commented on HTML5-frontend/script.js:
```diff
@@ -1,3 +1,6 @@
+if (typeof simulate === 'undefined') {
```
What is this?

mtlynch commented:
Reviewed 3 of 8 files at r1.
Review status: 3 of 8 files reviewed at latest revision, 8 unresolved discussions.

---

*[HTML5-frontend/ndt-wrapper.js, line 59 \[r1\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaDDJrXrId-qRSb8VF:-KCaDDJrXrId-qRSb8VG:-144416865) ([raw file](https://github.com/m-lab/ndt/blob/0a78966f643fd2b35d02804d6223e62244b2a612/HTML5-frontend/ndt-wrapper.js#L59)):*
Can we document what basePath is?

---

*[HTML5-frontend/ndt-wrapper.js, line 123 \[r1\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaDIDClGWglDEso_C1:-KCaDIDClGWglDEso_C2:-1129537454) ([raw file](https://github.com/m-lab/ndt/blob/0a78966f643fd2b35d02804d6223e62244b2a612/HTML5-frontend/ndt-wrapper.js#L123)):*
Do we have to save `this.client`? Could we simplify to:

```
(new NDTjs(this._hostname, this._port, this._path, this.callbacks, this._update_interval)).startTest();

---

*[HTML5-frontend/ndt-wrapper.js, line 127 \[r1\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaDTKEFFM9ZBCh_VNy:-KCaDTKFUmVcR_RsJ4rz:-1258847200) ([raw file](https://github.com/m-lab/ndt/blob/0a78966f643fd2b35d02804d6223e62244b2a612/HTML5-frontend/ndt-wrapper.js#L127)):*
We probably shouldn't even be in this function if we failed to find a suitable NDT server. If we're here and `this._hostname` is null, it means the calling code is wrong, so suggest re-wording to something like "No server hostname specified. Aborting."

Also, this is a style preference, so it's up to you, but I generally find code more readable when it's structured like:

```
if (!this._hostname) {
  console.log('whatever');
  return false;
}
this.client = new NDTjs(this._hostname, this._port, this._path,
                        this.callbacks, this._update_interval);
this.client.startTest();
```

---

*[HTML5-frontend/script.js, line 1 \[r1\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaEObW3HToL_BEmUCo:-KCaEObXysV6ilAhesGc:168090562) ([raw file](https://github.com/m-lab/ndt/blob/0a78966f643fd2b35d02804d6223e62244b2a612/HTML5-frontend/script.js#L1)):*
What is this?

---

*[HTML5-frontend/script.js, line 35 \[r1\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaH2CuADbHVlRoBOK5:-KCaH2CuADbHVlRoBOK6:519404090) ([raw file](https://github.com/m-lab/ndt/blob/0a78966f643fd2b35d02804d6223e62244b2a612/HTML5-frontend/script.js#L35)):*
Can we eliminate mlab-ns entirely? I'd prefer it to be the frontend's responsibility to find the NDT host. The default frontend in this repo (which fakewww will server) can assume that the host is the same as the web server.

---

*[HTML5-frontend/script.js, line 80 \[r1\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaJK4jzZLbznVy-LwV:-KCaJK4kDjmsM1m9iavB:753395594) ([raw file](https://github.com/m-lab/ndt/blob/627754a4e2b8054b55e386f42bb882226e5f3585/HTML5-frontend/script.js#L80)):*
Why this change?

---

*[HTML5-frontend/script.js, line 86 \[r1\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaFbmv8eYBBMjnVFt4:-KCaFbmv8eYBBMjnVFt5:1377630708) ([raw file](https://github.com/m-lab/ndt/blob/0a78966f643fd2b35d02804d6223e62244b2a612/HTML5-frontend/script.js#L86)):*
What happens if it's an empty string?

---

*[HTML5-frontend/script.js, line 654 \[r1\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaGngZ7N0Gx_3B4Iov:-KCaGngZ7N0Gx_3B4Iow:-34503613) ([raw file](https://github.com/m-lab/ndt/blob/0a78966f643fd2b35d02804d6223e62244b2a612/HTML5-frontend/script.js#L654)):*
Should this be `!==`?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/m-lab/ndt/46)*
<!-- Sent from Reviewable.io -->

mtlynch commented:
Reviewed 5 of 8 files at r1.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/m-lab/ndt/46)*
<!-- Sent from Reviewable.io -->

nkinkade commented:
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[HTML5-frontend/ndt-wrapper.js, line 59 \[r1\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaDDJrXrId-qRSb8VF:-KCaR8nSesA-RFdUkFFa:-499146009) ([raw file](https://github.com/m-lab/ndt/blob/0a78966f643fd2b35d02804d6223e62244b2a612/HTML5-frontend/ndt-wrapper.js#L59)):*
Done.

---

*[HTML5-frontend/ndt-wrapper.js, line 123 \[r1\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaDIDClGWglDEso_C1:-KCaQCAo2YEZHgrGh40v:-250168119) ([raw file](https://github.com/m-lab/ndt/blob/0a78966f643fd2b35d02804d6223e62244b2a612/HTML5-frontend/ndt-wrapper.js#L123)):*
This was original code.  I didn't this.

---

*[HTML5-frontend/ndt-wrapper.js, line 127 \[r1\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaDTKEFFM9ZBCh_VNy:-KCaRFbd5tyLN3sfVI2I:405062603) ([raw file](https://github.com/m-lab/ndt/blob/0a78966f643fd2b35d02804d6223e62244b2a612/HTML5-frontend/ndt-wrapper.js#L127)):*
I put it here, which is rather late in the whole process, as some small effort to make sure the mlab-ns request has completed, even though I've set the $.ajax() function which calls mlab-ns to run synchronously.

---

*[HTML5-frontend/script.js, line 1 \[r1\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaEObW3HToL_BEmUCo:-KCaQYqitBUl3qkGWBj_:1315844977) ([raw file](https://github.com/m-lab/ndt/blob/0a78966f643fd2b35d02804d6223e62244b2a612/HTML5-frontend/script.js#L1)):*
I have no idea what that is.  It was in the original code.

---

*[HTML5-frontend/script.js, line 35 \[r1\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaH2CuADbHVlRoBOK5:-KCaQlNVVyRlt-EiXWmU:1285289724) ([raw file](https://github.com/m-lab/ndt/blob/0a78966f643fd2b35d02804d6223e62244b2a612/HTML5-frontend/script.js#L35)):*
We could, but it's worth noting that there was already a findNdtServer() function in the NDT codebase before, it just wasn't being used.  At the end of the day, M-Lab is probably one of the few people/orgs in the world who uses the NDT codebase, so it doesn't seem unreasonable to just bake in some amount of M-Labness.

---

*[HTML5-frontend/script.js, line 80 \[r1\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaJK4jzZLbznVy-LwV:-KCaRylLgkAa7o9MtrdY:529343840) ([raw file](https://github.com/m-lab/ndt/blob/627754a4e2b8054b55e386f42bb882226e5f3585/HTML5-frontend/script.js#L80)):*
I didn't change anything here. If there was a change it must have been whitespace.

---

*[HTML5-frontend/script.js, line 86 \[r1\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaFbmv8eYBBMjnVFt4:-KCaQvkwrgecg5AmSsNw:1341285902) ([raw file](https://github.com/m-lab/ndt/blob/0a78966f643fd2b35d02804d6223e62244b2a612/HTML5-frontend/script.js#L86)):*
It won't be.  Earlier in the file I check for its existence, and if it doesn't exist it gets set to an empty string.

---

*[HTML5-frontend/script.js, line 654 \[r1\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaGngZ7N0Gx_3B4Iov:-KCaSSptRCSdxX94G-Bp:-499146009) ([raw file](https://github.com/m-lab/ndt/blob/0a78966f643fd2b35d02804d6223e62244b2a612/HTML5-frontend/script.js#L654)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/m-lab/ndt/46)*
<!-- Sent from Reviewable.io -->

mtlynch commented:
Reviewed 1 of 2 files at r2.
Review status: 7 of 8 files reviewed at latest revision, 7 unresolved discussions.

---

*[HTML5-frontend/ndt-wrapper.js, line 127 \[r1\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaDTKEFFM9ZBCh_VNy:-KCaV6nd1JxKfMhzUcZ-:49553882) ([raw file](https://github.com/m-lab/ndt/blob/0a78966f643fd2b35d02804d6223e62244b2a612/HTML5-frontend/ndt-wrapper.js#L127)):*
Right, I'm not objecting to the check, just the wording of the error message.

---

*[HTML5-frontend/ndt-wrapper.js, line 60 \[r2\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaW1Hi-Ls3W5l9cimH:-KCaW1Hjl0aD_R4ySCiX:1434178682) ([raw file](https://github.com/m-lab/ndt/blob/62f90510527762aa7a504913188a556d26136c5f/HTML5-frontend/ndt-wrapper.js#L60)):*
Can the caller specify the full path to the script? Otherwise, every client would have to create a file called ndt-wrapper-ww.js, right?

---

*[HTML5-frontend/ndt-wrapper.js, line 61 \[r2\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaVWwTs0WPsIoHTg2l:-KCaVWwTs0WPsIoHTg2m:450544914) ([raw file](https://github.com/m-lab/ndt/blob/62f90510527762aa7a504913188a556d26136c5f/HTML5-frontend/ndt-wrapper.js#L61)):*
Anything JS script can be the caller, not just `script.js` right?

---

*[HTML5-frontend/script.js, line 1 \[r1\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaEObW3HToL_BEmUCo:-KCaTubVs_SkJ5Q3jVxf:-2044726912) ([raw file](https://github.com/m-lab/ndt/blob/0a78966f643fd2b35d02804d6223e62244b2a612/HTML5-frontend/script.js#L1)):*
Is this a move? The diff shows it as new code. Can you see it in the diff as new code?

---

*[HTML5-frontend/script.js, line 35 \[r1\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaH2CuADbHVlRoBOK5:-KCaTzkuYt8Hzk7daYyr:850962193) ([raw file](https://github.com/m-lab/ndt/blob/0a78966f643fd2b35d02804d6223e62244b2a612/HTML5-frontend/script.js#L35)):*
I'd rather get rid of it entirely. I think it adds unnecessary complexity and the `if port == 7123` thing is not a very robust check, so I'd prefer if we didn't rely on that.

---

*[HTML5-frontend/script.js, line 80 \[r1\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaJK4jzZLbznVy-LwV:-KCaUMFZSRrMCZ79laQt:-331652028) ([raw file](https://github.com/m-lab/ndt/blob/627754a4e2b8054b55e386f42bb882226e5f3585/HTML5-frontend/script.js#L80)):*
Diff shows that you deleted a "return" entirely. Not whitespace.

---

*[HTML5-frontend/script.js, line 86 \[r1\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaFbmv8eYBBMjnVFt4:-KCaUFBw5HHLZWCGQuYf:46147688) ([raw file](https://github.com/m-lab/ndt/blob/0a78966f643fd2b35d02804d6223e62244b2a612/HTML5-frontend/script.js#L86)):*
I don't understand what you mean. If you set it to an empty string, then it will be an empty string. What happens then?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/m-lab/ndt/46)*
<!-- Sent from Reviewable.io -->

nkinkade commented:
Review status: 7 of 8 files reviewed at latest revision, 7 unresolved discussions.

---

*[HTML5-frontend/ndt-wrapper.js, line 127 \[r1\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaDTKEFFM9ZBCh_VNy:-KCabtsMJs7UqGOYwEP8:-248636053) ([raw file](https://github.com/m-lab/ndt/blob/0a78966f643fd2b35d02804d6223e62244b2a612/HTML5-frontend/ndt-wrapper.js#L127)):*
Since I've removed the mlab-ns code, there will never be a case where it will be empty, unless the value of _location.hostname_ is empty.  Removing this check.

---

*[HTML5-frontend/ndt-wrapper.js, line 60 \[r2\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaW1Hi-Ls3W5l9cimH:-KCaaPJDEEhcA8jTqHgh:129846828) ([raw file](https://github.com/m-lab/ndt/blob/62f90510527762aa7a504913188a556d26136c5f/HTML5-frontend/ndt-wrapper.js#L60)):*
Yes, but we're not talking about _every client_, we are specifically talking about the code in the NDT repository, which we apparently wanted to reuse, only changing the HTML frontend.

---

*[HTML5-frontend/ndt-wrapper.js, line 61 \[r2\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaVWwTs0WPsIoHTg2l:-KCaa2Ruuepu3D66mbdz:-1102869020) ([raw file](https://github.com/m-lab/ndt/blob/62f90510527762aa7a504913188a556d26136c5f/HTML5-frontend/ndt-wrapper.js#L61)):*
Sure, something other than script.js could call this file, but that is not how the code currently works. My understanding was that our goal here was to keep the code in the NDT repository working as it, but with the flexibility to add new HTML frontends. 

---

*[HTML5-frontend/script.js, line 1 \[r1\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaEObW3HToL_BEmUCo:-KCaXNi0724lOGLqlq4c:1802648480) ([raw file](https://github.com/m-lab/ndt/blob/0a78966f643fd2b35d02804d6223e62244b2a612/HTML5-frontend/script.js#L1)):*
I put that typeof wrapper around a code block that already existed.  The reason is that the existing widget.html file [defined this variable](https://github.com/m-lab/ndt/blob/master/HTML5-frontend/widget.html#L17), but we don't define it in our HTML, and others might not as well.  My goal in the changeset was to _not_ remove any existing functionality, but simply to workaround it.

---

*[HTML5-frontend/script.js, line 35 \[r1\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaH2CuADbHVlRoBOK5:-KCaZFqJGw_WDDXZaeG2:-499146009) ([raw file](https://github.com/m-lab/ndt/blob/0a78966f643fd2b35d02804d6223e62244b2a612/HTML5-frontend/script.js#L35)):*
Done.

---

*[HTML5-frontend/script.js, line 80 \[r1\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaJK4jzZLbznVy-LwV:-KCa_dwMe8_mJkwzs925:-102599142) ([raw file](https://github.com/m-lab/ndt/blob/627754a4e2b8054b55e386f42bb882226e5f3585/HTML5-frontend/script.js#L80)):*
Oh, Reviewable wasn't show that change to me for some reason.  Probably just me not knowing how to use Reviewable.  In any case, I had removed that return statement because it seems like left-over debugging thing.  I added it back.

---

*[HTML5-frontend/script.js, line 86 \[r1\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaFbmv8eYBBMjnVFt4:-KCaZbE_z6Zab6cds_V3:-420519020) ([raw file](https://github.com/m-lab/ndt/blob/0a78966f643fd2b35d02804d6223e62244b2a612/HTML5-frontend/script.js#L86)):*
```new Worker``` [is called](https://github.com/nkinkade/ndt/blob/decouple-ws-html/HTML5-frontend/ndt-wrapper.js#L72) like this ```this.worker = new Worker(basePath + 'ndt-wrapper-ww.js');``` If _basePath_ is empty then it's just trying to load ndt-wrapper-ww.js from the baseURI of the site.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/m-lab/ndt/46)*
<!-- Sent from Reviewable.io -->

mtlynch commented:
Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[HTML5-frontend/ndt-wrapper.js, line 60 \[r2\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaW1Hi-Ls3W5l9cimH:-KCaw4zo4X2ZJP2yqmt7:1702796895) ([raw file](https://github.com/m-lab/ndt/blob/62f90510527762aa7a504913188a556d26136c5f/HTML5-frontend/ndt-wrapper.js#L60)):*
Ah, okay, I had misunderstood and thought that other frontends (like the M-Lab web site) would have to write their own ndt-wrapper-ww.js file and give it that specific name.

So if I'm understanding correctly, we want all clients to use this specific file and the only reason that we're parameterizing the path at all is that JavaScript prevents us from referencing the file relative to our current location, correct? If so, can we spare clients the pain of having to locate this file by detecting where we are automatically?

```javascript
var scriptPath = function () {
  var scripts = document.getElementsByTagName('SCRIPT');
  var path = '';
    if (scripts && scripts.length > 0) {
      for(var i in scripts) {
          if(scripts[i].src && scripts[i].src.match(/\/ndt-wrapper\.js$/)) {
              path = scripts[i].src.replace(/(.*)\/ndt-wrapper\.js$/, '$1');
                  break;          
                    }                   
            }         
    }
    return path.substring(location.origin.length);
};
```
(adapted from [here](http://www.bencarpenter.co.uk/javascript-path-to-the-current-script))

---

*[HTML5-frontend/ndt-wrapper.js, line 61 \[r2\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaVWwTs0WPsIoHTg2l:-KCaza0fe-4TLq3WjBGT:-1107073237) ([raw file](https://github.com/m-lab/ndt/blob/62f90510527762aa7a504913188a556d26136c5f/HTML5-frontend/ndt-wrapper.js#L61)):*
I think just for the sake of loose coupling, functions shouldn't make assumptions about who their callers are, so it's sufficient to just say "the caller"

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/m-lab/ndt/46)*
<!-- Sent from Reviewable.io -->

nkinkade commented:
Review status: 6 of 8 files reviewed at latest revision, 3 unresolved discussions.

---

*[HTML5-frontend/ndt-wrapper.js, line 60 \[r2\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaW1Hi-Ls3W5l9cimH:-KCbEvloVyS42U_fNC2Y:-1802794310) ([raw file](https://github.com/m-lab/ndt/blob/62f90510527762aa7a504913188a556d26136c5f/HTML5-frontend/ndt-wrapper.js#L60)):*
Yes, that is the idea, and thanks for that code snippet.  Interestingly, I had already started to go down that path [and had something similar](https://github.com/m-lab/ndt/commit/10c1e1c8ae1c1263339490d02948b4ec8438f2c0#diff-beb42ae6787c790c9e830bdf2af2228eR30), though not as sophisticated.  For some reason my solution didn't work all the time so I abandoned it in favor of a manually set variable.  I've implemented the code snippet you pasted, but with a few modifications.  Hopefully this is robust.

---

*[HTML5-frontend/ndt-wrapper.js, line 61 \[r2\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCaVWwTs0WPsIoHTg2l:-KCbFaezm0igb9j6uqNJ:298210887) ([raw file](https://github.com/m-lab/ndt/blob/62f90510527762aa7a504913188a556d26136c5f/HTML5-frontend/ndt-wrapper.js#L61)):*
I've updated the comment to be more generic.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/m-lab/ndt/46)*
<!-- Sent from Reviewable.io -->

mtlynch commented:
Review status: 6 of 8 files reviewed at latest revision, 7 unresolved discussions.

---

*[HTML5-frontend/ndt-wrapper.js, line 59 \[r4\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCbhtJSY9-zDSj4tM9U:-KCbhtJT3OtFYMD-j_yi:1081517993) ([raw file](https://github.com/m-lab/ndt/blob/ce331f96b333d23374c96ddfa270d4de4627cabc/HTML5-frontend/ndt-wrapper.js#L59)):*
This no longer needs a parameter, right?

---

*[HTML5-frontend/ndt-wrapper.js, line 67 \[r4\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCbgzPMmICpialC7BSE:-KCbgzPNPAKXs8ZhqMUD:1327365105) ([raw file](https://github.com/m-lab/ndt/blob/ce331f96b333d23374c96ddfa270d4de4627cabc/HTML5-frontend/ndt-wrapper.js#L67)):*
Can we replace tabs with spaces?

---

*[HTML5-frontend/script.js, line 35 \[r4\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCbi_o_mR_hwbufeERA:-KCbi_oaX7JtIJz86am0:-1290012144) ([raw file](https://github.com/m-lab/ndt/blob/ce331f96b333d23374c96ddfa270d4de4627cabc/HTML5-frontend/script.js#L35)):*
This is now unused, right?

---

*[HTML5-frontend/script.js, line 45 \[r4\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCbie7-hwKJkQKwlsTz:-KCbie7-hwKJkQKwlsU-:-1677837446) ([raw file](https://github.com/m-lab/ndt/blob/ce331f96b333d23374c96ddfa270d4de4627cabc/HTML5-frontend/script.js#L45)):*
ditto?

---

*[HTML5-frontend/script.js, line 666 \[r4\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCbhESQGsdnFyfkO07f:-KCbhESRvtFp1i7zVU_r:1080924333) ([raw file](https://github.com/m-lab/ndt/blob/ce331f96b333d23374c96ddfa270d4de4627cabc/HTML5-frontend/script.js#L666)):*
Can we document `scriptName`?

---

*[HTML5-frontend/script.js, line 669 \[r4\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCbhI6LIqATqrIHqySm:-KCbhI6MlJKxMSeV26Vo:-45079789) ([raw file](https://github.com/m-lab/ndt/blob/ce331f96b333d23374c96ddfa270d4de4627cabc/HTML5-frontend/script.js#L669)):*
The caller needs to either pass something that's already a regex or we need to escape the regex here. As currently implemented, the `.` before the extension is being interpreted as a wildcard.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/m-lab/ndt/46)*
<!-- Sent from Reviewable.io -->

nkinkade commented:
Review status: 6 of 8 files reviewed at latest revision, 7 unresolved discussions.

---

*[HTML5-frontend/ndt-wrapper.js, line 59 \[r4\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCbhtJSY9-zDSj4tM9U:-KCbmA_evm-dD6FVyV4j:1184614911) ([raw file](https://github.com/m-lab/ndt/blob/ce331f96b333d23374c96ddfa270d4de4627cabc/HTML5-frontend/ndt-wrapper.js#L59)):*
Good catch.  Removed.

---

*[HTML5-frontend/ndt-wrapper.js, line 67 \[r4\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCbgzPMmICpialC7BSE:-KCbmQGQy5aGycvhIOKC:838833015) ([raw file](https://github.com/m-lab/ndt/blob/ce331f96b333d23374c96ddfa270d4de4627cabc/HTML5-frontend/ndt-wrapper.js#L67)):*
Sorry about that.  For my own stuff I generally use tabs, and just forgot to set expandtab in vim.  Fixed.

---

*[HTML5-frontend/script.js, line 35 \[r4\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCbi_o_mR_hwbufeERA:-KCbmoHqs-F42xg5GuG0:-1892019429) ([raw file](https://github.com/m-lab/ndt/blob/ce331f96b333d23374c96ddfa270d4de4627cabc/HTML5-frontend/script.js#L35)):*
No, [it's still used in ndt-wrapper.js](https://github.com/nkinkade/ndt/blob/decouple-ws-html/HTML5-frontend/ndt-wrapper.js#L33).

---

*[HTML5-frontend/script.js, line 45 \[r4\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCbie7-hwKJkQKwlsTz:-KCbn3iWLp_ZfhTpC3in:-1269420314) ([raw file](https://github.com/m-lab/ndt/blob/ce331f96b333d23374c96ddfa270d4de4627cabc/HTML5-frontend/script.js#L45)):*
Yeah, that is no longer needed.  Removed.

---

*[HTML5-frontend/script.js, line 666 \[r4\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCbhESQGsdnFyfkO07f:-KCbnRSxIB8oeOEjVk6F:-499146009) ([raw file](https://github.com/m-lab/ndt/blob/ce331f96b333d23374c96ddfa270d4de4627cabc/HTML5-frontend/script.js#L666)):*
Done.

---

*[HTML5-frontend/script.js, line 669 \[r4\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCbhI6LIqATqrIHqySm:-KCbnZQE8jY0y_zWkKDY:-1504573269) ([raw file](https://github.com/m-lab/ndt/blob/ce331f96b333d23374c96ddfa270d4de4627cabc/HTML5-frontend/script.js#L669)):*
True, but I don't think it has any practical effect.  Any passed dot will match itself, so it will still work as expected.  Can you think of a case where not escaping the input would produce some sort of error condition?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/m-lab/ndt/46)*
<!-- Sent from Reviewable.io -->

mtlynch commented:
Reviewed 2 of 2 files at r5.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[HTML5-frontend/script.js, line 35 \[r4\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCbi_o_mR_hwbufeERA:-KCbroNeCrJrQkUG03HW:1771546430) ([raw file](https://github.com/m-lab/ndt/blob/ce331f96b333d23374c96ddfa270d4de4627cabc/HTML5-frontend/script.js#L35)):*
Can we pass it explicitly alongside the server parameter? This is essentially passing it as a global, right?

---

*[HTML5-frontend/script.js, line 669 \[r4\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCbhI6LIqATqrIHqySm:-KCbqNKhozwm6UCUhbNb:919035604) ([raw file](https://github.com/m-lab/ndt/blob/ce331f96b333d23374c96ddfa270d4de4627cabc/HTML5-frontend/script.js#L669)):*
Doesn't have a practical effect with that string, but I really don't like the idea of knowingly failing to encode an input properly. It's unexpected from the caller's perspective that the string will be blindly interpreted as a regex, so there's potential for maintenance "surprises."

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/m-lab/ndt/46)*
<!-- Sent from Reviewable.io -->

nkinkade commented:
Review status: 5 of 9 files reviewed at latest revision, 4 unresolved discussions.

---

*[HTML5-frontend/script.js, line 35 \[r4\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCbi_o_mR_hwbufeERA:-KCbv9dILzVMFlDTcdrB:-941888294) ([raw file](https://github.com/m-lab/ndt/blob/ce331f96b333d23374c96ddfa270d4de4627cabc/HTML5-frontend/script.js#L35)):*
I've removed the global variable and now the protocol variable gets passed explicitly to everything as a parameter.

---

*[HTML5-frontend/script.js, line 669 \[r4\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCbhI6LIqATqrIHqySm:-KCbsnvcyawohDhK4mKJ:-1946923628) ([raw file](https://github.com/m-lab/ndt/blob/ce331f96b333d23374c96ddfa270d4de4627cabc/HTML5-frontend/script.js#L669)):*
What is an example of any (valid) string that could be passed which would fail? The only valid strings to be passed in _scriptName_ are bare file names, as the comment describes.  If someone passes an invalid string or non-existent file, then the check is going to fail anyway. It seems to me that this is one of those cases where we'd just be adding additional lines of code and processing for what amounts to a non-issue.  That a dot matches itself is not invalid, whether the user expects that or not.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/m-lab/ndt/46)*
<!-- Sent from Reviewable.io -->

mtlynch commented:
Reviewed 4 of 4 files at r6.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[HTML5-frontend/script.js, line 35 \[r4\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCbi_o_mR_hwbufeERA:-KCc5OZp8UfLxtQhC-gZ:1619823185) ([raw file](https://github.com/m-lab/ndt/blob/ce331f96b333d23374c96ddfa270d4de4627cabc/HTML5-frontend/script.js#L35)):*
Thanks, this is much clearer.

---

*[HTML5-frontend/script.js, line 669 \[r4\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCbhI6LIqATqrIHqySm:-KCc5yNgMDH2Xf9DasF3:-689506878) ([raw file](https://github.com/m-lab/ndt/blob/ce331f96b333d23374c96ddfa270d4de4627cabc/HTML5-frontend/script.js#L669)):*
>What is an example of any (valid) string that could be passed which would fail?

A script called `ndt+ssl.js` would break this.

If we change the documentation to just say that it takes a regex instead of a bare filename, we don't need to add any code. I'd also be fine with just making this like `findNdtWrapper` and hardcoding the filename within the function.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/m-lab/ndt/46)*
<!-- Sent from Reviewable.io -->

nkinkade commented:
Review status: 7 of 9 files reviewed at latest revision, 2 unresolved discussions.

---

*[HTML5-frontend/script.js, line 669 \[r4\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCbhI6LIqATqrIHqySm:-KCcO1wORS2FGF5IiKi5:-1603600800) ([raw file](https://github.com/m-lab/ndt/blob/ce331f96b333d23374c96ddfa270d4de4627cabc/HTML5-frontend/script.js#L669)):*
Okay, you're right, and your original code snippet had 'ndt-wrapper.js' hardcoded anyway.  I had just thought I would try to make it more flexible, but nothing about this entire thing is flexible anyway.  I've added a commit which just hardcodes the script name.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/m-lab/ndt/46)*
<!-- Sent from Reviewable.io -->

mtlynch commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 7 of 9 files reviewed at latest revision, 1 unresolved discussion.

---

*[HTML5-frontend/script.js, line 660 \[r7\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCpcPqNwa3jUYVSEV_q:-KCpcPqOz1DsoU8Ux__m:-1315601668) ([raw file](https://github.com/m-lab/ndt/blob/af50ca9d32bc902d64cdc5af878e020e4ab2c1a1/HTML5-frontend/script.js#L660)):*
We need to update the documentation

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/m-lab/ndt/46#-:-KCpcXylmApHQjcVJ1RK:1417088721)*
<!-- Sent from Reviewable.io -->

nkinkade commented:
Review status: 7 of 9 files reviewed at latest revision, 1 unresolved discussion.

---

*[HTML5-frontend/script.js, line 660 \[r7\]](https://reviewable.io:443/reviews/m-lab/ndt/46#-KCpcPqNwa3jUYVSEV_q:-KCpxoa6mfL68SehJjhz:-499146009) ([raw file](https://github.com/m-lab/ndt/blob/af50ca9d32bc902d64cdc5af878e020e4ab2c1a1/HTML5-frontend/script.js#L660)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/m-lab/ndt/46)*
<!-- Sent from Reviewable.io -->
